### ISSUE TITLE
Gateway application fails when creating from JDL with multiple apps

### ISSUE THREAD

**Issue by deepu105** [jhipster/generator-jhipster#17324]
##### **Overview of the issue**

When running a [JDL](https://gist.github.com/deepu105/1f0663af2acdb67dd5472c4896b595cf) with multiple apps, the gateway generation fails with the below stack trace. Seems like an issue with reading `clientFramework`. This happens only on version 7.4.1, 7.4.0 works fine. 

```
jhipster jdl app.jdl --fork                                          
....

This is an existing project, using the configuration from your .yo-rc.json file 
to re-generate the project...

This is an existing project, using the configuration from your .yo-rc.json file 
to re-generate the project...

This is an existing project, using the configuration from your .yo-rc.json file 
to re-generate the project...


Found the .jhipster/Invoice.json configuration file, entity can be automatically generated!


Found the .jhipster/Shipment.json configuration file, entity can be automatically generated!

     info changelogDate is missing in .jhipster/Invoice.json, using 20211216151816 as fallback
     info changelogDate is missing in .jhipster/Shipment.json, using 20211216151916 as fallback
This is an existing project, using the configuration from your .yo-rc.json file 
to re-generate the project...


Found the .jhipster/Product.json configuration file, entity can be automatically generated!

     info Creating changelog for entities Invoice,Shipment

Found the .jhipster/ProductCategory.json configuration file, entity can be automatically generated!


Found the .jhipster/ProductOrder.json configuration file, entity can be automatically generated!


Found the .jhipster/OrderItem.json configuration file, entity can be automatically generated!

     info changelogDate is missing in .jhipster/Product.json, using 20211216151815 as fallback
     info changelogDate is missing in .jhipster/ProductCategory.json, using 20211216151915 as fallback
     info changelogDate is missing in .jhipster/ProductOrder.json, using 20211216152015 as fallback
     info changelogDate is missing in .jhipster/OrderItem.json, using 20211216152115 as fallback
     info Creating changelog for entities Product,ProductCategory,ProductOrder,OrderItem

Found the .jhipster/Notification.json configuration file, entity can be automatically generated!

     info changelogDate is missing in .jhipster/Notification.json, using 20211216151816 as fallback
ERROR! Cannot read properties of undefined (reading 'clientFramework')
TypeError: Cannot read properties of undefined (reading 'clientFramework')
    at EntityGenerator.setupMicroServiceEntity (/home/deepu/.npm-global/lib/node_modules/generator-jhipster/generators/entity/index.js:237:30)
    at Object.<anonymous> (/home/deepu/.npm-global/lib/node_modules/generator-jhipster/node_modules/yeoman-generator/lib/index.js:1024:25)
    at /home/deepu/.npm-global/lib/node_modules/generator-jhipster/node_modules/run-async/index.js:49:25
    at new Promise (<anonymous>)
    at /home/deepu/.npm-global/lib/node_modules/generator-jhipster/node_modules/run-async/index.js:26:19
    at runLoop.add.once.once (/home/deepu/.npm-global/lib/node_modules/generator-jhipster/node_modules/yeoman-generator/lib/index.js:1025:11)
    at Immediate.<anonymous> (/home/deepu/.npm-global/lib/node_modules/generator-jhipster/node_modules/grouped-queue/lib/subqueue.js:48:34)
    at processImmediate (node:internal/timers:464:21)
INFO! Generator app child process exited with code 1
ERROR! Error executing app --reproducible --no-force --with-entities --no-dry-run --no-whitespace --no-bail --no-install-path --no-skip-regenerate --no-skip-yo-resolve --from-jdl --no-skip-cache --no-skip-install --no-force-install --no-ask-answered --no-defaults --no-skip-git
Error: Error executing app --reproducible --no-force --with-entities --no-dry-run --no-whitespace --no-bail --no-install-path --no-skip-regenerate --no-skip-yo-resolve --from-jdl --no-skip-cache --no-skip-install --no-force-install --no-ask-answered --no-defaults --no-skip-git
    at ChildProcess.<anonymous> (/home/deepu/.npm-global/lib/node_modules/generator-jhipster/cli/import-jdl.js:195:16)
    at ChildProcess.emit (node:events:390:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
```

##### **Reproduce the error**

Run `jhipster jdl app.jdl --fork` with JDL mentioned above

##### **JHipster Version(s)**

7.4.1

##### **Browsers and Operating System**

Fedora

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by mshima** [comment]
Could not reproduce using main.

**Comment by AjayLaande** [comment]
I ran into the above issue with version 7.4.1, but it works on 7.4.0

The Gateway app generation fails with the below error/exception 

INFO! Using JHipster version installed globally
INFO! Executing import-jdl microservice-ecommerce-store-k8s.jdl
INFO! The JDL is being parsed.
INFO! Found entities: Product, ProductCategory, Customer, ProductOrder, OrderItem, Invoice, Shipment, Notification.
INFO! The JDL has been successfully parsed
INFO! Generating 3 applications.

Found the .jhipster/Shipment.json configuration file, entity can be automatically generated!

     info changelogDate is missing in .jhipster/Invoice.json, using 20220101115412 as fallback
     info changelogDate is missing in .jhipster/Shipment.json, using 20220101115512 as fallback
     info Creating changelog for entities Invoice,Shipment
ERROR! Cannot read properties of undefined (reading 'clientFramework')
TypeError: Cannot read properties of undefined (reading 'clientFramework')
    at EntityGenerator.setupMicroServiceEntity (/Users/ajay/nodejs/lib/node_modules/generator-jhipster/generators/entity/index.js:237:30)
    at Object.<anonymous> (/Users/ajay/nodejs/lib/node_modules/generator-jhipster/node_modules/yeoman-generator/lib/index.js:1024:25)
    at /Users/ajay/nodejs/lib/node_modules/generator-jhipster/node_modules/run-async/index.js:49:25
    at new Promise (<anonymous>)
    at /Users/ajay/nodejs/lib/node_modules/generator-jhipster/node_modules/run-async/index.js:26:19
    at runLoop.add.once.once (/Users/ajay/nodejs/lib/node_modules/generator-jhipster/node_modules/yeoman-generator/lib/index.js:1025:11)
    at Immediate.<anonymous> (/Users/ajay/nodejs/lib/node_modules/generator-jhipster/node_modules/grouped-queue/lib/subqueue.js:48:34)
    at processImmediate (node:internal/timers:464:21)
INFO! Generator app child process exited with code 1
ERROR! Error executing app --reproducible --no-force --with-entities --no-dry-run --no-whitespace --no-bail --no-install-path --no-skip-regenerate --no-skip-yo-resolve --from-jdl --no-skip-cache --skip-install --no-force-install --no-ask-answered --no-defaults --skip-git
Error: Error executing app --reproducible --no-force --with-entities --no-dry-run --no-whitespace --no-bail --no-install-path --no-skip-regenerate --no-skip-yo-resolve --from-jdl --no-skip-cache --skip-install --no-force-install --no-ask-answered --no-defaults --skip-git
    at ChildProcess.<anonymous> (/Users/ajay/nodejs/lib/node_modules/generator-jhipster/cli/import-jdl.js:195:16)
    at ChildProcess.emit (node:events:390:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)


Steps to reproduce

Generate the multi-app project with jdl file (from jhipster-jdl-samples repo)
`jhipster import-jdl --skip-install --skip-git microservice-ecommerce-store-k8s.jdl --fork`

Environment: 
Jhipster global version 7.4.1   
OS: Macos Mojave

**Comment by mshima** [comment]
Should be fixed in 7.5.0
